src/server/DeviceServer.js

```diff
@@ -87,9 +89,26 @@ class DeviceServer {
     this._eventPublisher = eventPublisher;
     this._areSystemFirmwareAutoupdatesEnabled =
       areSystemFirmwareAutoupdatesEnabled;
+
+    // Master listens to events from workers and broadcast them
+    // to all other workers except the owner.
+    if (useCluster && cluster.isMaster) {
```

Comment:
this is the main broadcasting hub.  I'm not sure whom this code belongs, too. I put this in the DeviceServer constructor, but may be EventPublisher constructor is a better place.